src/serializer/ResidualFunctions.js

```diff
@@ -174,22 +176,41 @@ export class ResidualFunctions {
     }
   }
 
-  _getReferentializedScopeInitialization(scope: ScopeBinding): BabelNodeIfStatement {
+  _getReferentializedScopeInitialization(
+    scope: ScopeBinding
+  ): [BabelNodeVariableDeclaration, BabelNodeIfStatement] | [BabelNodeIfStatement] {
     let properties = [];
     for (let [variableName, value] of scope.initializationValues.entries()) {
       properties.push(t.objectProperty(t.identifier(variableName), value));
     }
-
-    return t.ifStatement(
-      t.unaryExpression("!", t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true)),
-      t.expressionStatement(
-        t.assignmentExpression(
-          "=",
-          t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true),
-          t.objectExpression(properties)
-        )
-      )
-    );
+    let initExpression = t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true);
+    let capturedScope;
+    let capturedScopeId;
+    if (scope.capturedScope) {
+      capturedScope = scope.capturedScope;
+      capturedScopeId = t.identifier(capturedScope);
+
+      return [
+        t.variableDeclaration("var", [t.variableDeclarator(capturedScopeId, initExpression)]),
+        t.ifStatement(
+          t.unaryExpression("!", capturedScopeId),
+          t.expressionStatement(
+            t.assignmentExpression(
+              "=",
+              initExpression,
+              t.assignmentExpression("=", capturedScopeId, t.objectExpression(properties))
+            )
+          )
+        ),
+      ];
+    } else {
+      return [
```

Comment:
Is this `else` branch actually feasible? Is there a test case that exercises it? I'd expect it to be dead.